### ISSUE TITLE
:ObsidianTemplate fails to find templates in subdirectories

### ISSUE THREAD

**Issue by templeman** [epwalsh/obsidian.nvim#409]
### 🐛 Describe the bug

I have a bunch of templates that are further broken out into subdirectories:

```
Notes
    Templates
        sample-template.md
        category1
            template1.md
            template2.md
        category2
            template3.md
            template4.md
    .obsidian
    Note 1.md
    Note 2.md
```

Running `:ObsidianTemplate` from within an existing note brings up the picker with all of the template files listed, as expected:

```
sample-template.md
category1/template1.md
category1/template2.md
category2/template3.md
category2/template4.md
```

However, pressing `return` on any of the nested template files results in an error:

`ERROR Template file 'Users/sam/Dropbox (Personal)/Notes/Templates/template1.md' not found`

This error indicates that the template subdirectory (`category1/` in this case) isn't picked up. I can pick templates from the base template directory `Templates` and they do get inserted as expected.

Also these nested templates were working fine up until sometime within the last few weeks (maybe v3.0.0 but that's just a guess). A simple fix is to move all templates under the top-level `Templates` directory but I wanted to flag as this seems like a possible bug.

### Config

```
  {
    "epwalsh/obsidian.nvim",
    version = "*",
    lazy = true,
    event = {
      "BufReadPre " .. vim.fn.expand("~") .. "/Dropbox (Personal)/Notes/**.md",
      "BufNewFile " .. vim.fn.expand("~") .. "/Dropbox (Personal)/Notes/**.md",
    },
    -- stylua: ignore
    keys = {
      { '<Leader>nv', '<Cmd>ObsidianSearch<CR>', desc = 'obsidian: search', },
      { '<Leader>os', '<Cmd>ObsidianQuickSwitch<CR>', desc = 'obsidian: quick switch', },
    },
    dependencies = {
      "nvim-lua/plenary.nvim",
    },
    config = function()
      require("obsidian").setup({
        workspaces = {
          { name = "Notes", path = "~/Dropbox (Personal)/Notes" },
        },
        -- Optional, for templates (see below).
        templates = {
          subdir = "Templates",
        },
        daily_notes = {
          -- Optional, if you keep daily notes in a separate directory.
          folder = "journal/daily",
          -- Optional, if you want to automatically insert a template from your template directory like 'daily.md'
          template = "Temporal/journal-nvim.md",
        },
      })
    end,
  },
```

### Environment

```
nvim --version
NVIM v0.10.0-dev-2039+g583c1de17-Homebrew
Build type: Release
LuaJIT 2.1.1703358377

nvim --headless -c 'lua require("obsidian").info()' -c q
[obsidian.nvim (v3.3.0)] Commit SHA: e8c9f099b2829239b1cf05f65dafb80efe31e37b
[plenary.nvim] Commit SHA: 4f71c0c4a196ceb656c824a70792f3df3ce6bb6d
[fzf.vim] Commit SHA: 8ea9574b3ebdc44fbe3dc6381a13f9ce0c290e2f
ripgrep 14.1.0 (rev e50df40a19)%                                                                                                                                                                                  

:ObsidianDebug
Current buffer directory: '/Users/sam/Dropbox (Personal)/Notes'
Workspaces:
  Active workspace: Workspace(name='Notes', path='/Users/sam/Dropbox (Personal)/Notes', root='/Users/sam/Dropbox (Personal)/Notes')
```

**Comment by epwalsh** [comment]
Definitely a bug @templeman, I’ll TAL later today

**Comment by epwalsh** [comment]
Fixed in a83a958d019021b8d8d4b2372f0142e74eb348c7.
